code/game/objects/structures/floodlight.dm

```diff
@@ -0,0 +1,320 @@
+/obj/machinery/power/floodlight
+	name = "floodlight"
+	icon = 'icons/obj/floodlight.dmi'
+	icon_state = "floodlight0_hatchopen0"
+	desc = "An industrial floodlight. It has units of power remaining."
+	density = 1
+	anchored = 0
+	var/status = 0
+	var/obj/item/weapon/stock_parts/cell/powerpack
+	var/datum/effect_system/spark_spread/sparks = new
+	var/health = 60
+	var/maxhealth = 60
+	var/broken = 0
+	var/cover = 0
+	var/hascell = 0
+	var/brightness_on = 8
+	var/rigged = 0
+	var/wiredtoground = 0
+	use_power = 0
+	idle_power_usage = 0
+	active_power_usage = 40
+	req_access = list(access_engine_equip)
+	var/locked = 0
+	var/charge = 0
+	var/rawcharge = 0
+
+/obj/machinery/power/floodlight/New()
+	sparks.set_up(2, 0, src)
+	sparks.attach(src)
+	powerpack = null
+	turnoff()
+	update_health()
+	desc = "An industrial floodlight. It has no power source installed!"
+	SSobj.processing |= src
+	..()
+
+/obj/machinery/power/floodlight/Destroy()
+	qdel(sparks)
+	sparks = null
+	SSobj.processing.Remove(src)
+	return ..()
+
+/obj/machinery/power/floodlight/process()
+	if(broken)
+		return
+	if(powerpack == null)
+		hascell = 0
+	if(status == 0)
+		desc = "An industrial floodlight. Its off."
```

Comment:
This kinda iffy. Having this assigning isn't necessary and might add unwanted lag, but we're talking very very small amount of lag. Still, you could set the desc in turnOn() or turnOff()